keps/sig-network/3453-minimize-iptables-restore/README.md

```diff
@@ -0,0 +1,468 @@
+
+<!--
+Describe the metrics themselves and the reasons why they weren't added (e.g., cost,
+implementation difficulties, etc.).
+-->
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No (other than the things kube-proxy already depends on).
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No; this affects how kube-proxy processes the data it receives from
+the apiserver, but does not affect its communications with the
+apiserver at all.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No. The goal of the feature is specifically to _decrease_ the time
+take by operations related to kube-proxy SLIs/SLOs.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+There should be no effect on disk or IO.
+
+There should be a major _decrease_ in CPU usage by kube-proxy's
+process group (not by kube-proxy itself, but by the `iptables-restore`
+processes it spawns.)
+
+The current PR would not result in any non-negligible increase in
+kube-proxy RAM usage, although some variations of it might involve
+keeping track of more information in memory between resyncs.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+It does not change kube-proxy's behavior in this case.
+
+###### What are other known failure modes?
+
+(No _known_ failure modes yet. See [Risks and
+Mitigations](#risks-and-mitigations) for theorized potential failure
+modes.)
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+
+- 2022-08-02: Initial proposal
+
+## Drawbacks
+
+Assuming the code is not buggy, there are no drawbacks. The new code
+would be strictly superior to the old code.
```

Comment:
I don't think this should block the KEP at all, but if someone is touching these rules outside of kube-proxy we're more likely to clobber and correct them frequently without this change.